ptf_tests/tests/ct_rule_del_add_test.py

```diff
@@ -0,0 +1,156 @@
+from common.utils.gnmi_cli_utils import gnmi_cli_set_and_verify, gnmi_set_params, ip_set_ipv4
+
+
+class Connection_Track(BaseTest):
+
+    def setUp(self):
+        BaseTest.setUp(self)
+        self.result = unittest.TestResult()
+        config["relax"] = True # for verify_packets to ignore other packets received at the interface
+        
+        test_params = test_params_get()
+        config_json = test_params['config_json']
+        self.dataplane = ptf.dataplane_instance
+        ptf.dataplane_instance = ptf.dataplane.DataPlane(config)
+        self.capture_port = test_params['pci_bdf'][:-1] + "1"
+        self.config_data = get_config_dict(config_json, test_params['pci_bdf'])
+        self.gnmicli_params = get_gnmi_params_simple(self.config_data)
+        self.interface_ip_list = get_interface_ipv4_dict(self.config_data)
+
+
+    def runTest(self):
+        if not test_utils.gen_dep_files_p4c_dpdk_pna_ovs_pipeline_builder(self.config_data):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to generate P4C artifacts or pb.bin")
+        
+        if not gnmi_cli_set_and_verify(self.gnmicli_params):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to configure gnmi cli ports")
+
+        ip_set_ipv4(self.interface_ip_list)
+
+        # get port list and add to dataplane
+        port_list = self.config_data['port_list']
+        port_list[0] = test_utils.get_port_name_from_pci_bdf(self.capture_port)
+        port_ids = test_utils.add_port_to_dataplane(port_list)
+        
+        for port_id, ifname in config["port_map"].items():
+            device, port = port_id
+            self.dataplane.port_add(ifname, device, port)
+
+        if not ovs_p4ctl.ovs_p4ctl_set_pipe(self.config_data['switch'], self.config_data['pb_bin'], self.config_data['p4_info']):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to set pipe")
+
+        table = self.config_data['table'][0]
+        print(f"Rule Creation : {table['description']}")
+        print(f"Adding {table['description']} rules")
+        for match_action in table['match_action']:
+            if not ovs_p4ctl.ovs_p4ctl_add_entry(table['switch'],table['name'], match_action):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add table entry {match_action}")
+ 
+        table = self.config_data['table'][1]
+        print(f"Rule Creation : {table['description']}")
+        print(f"Adding {table['description']} rules")
+        for match_action in table['match_action']:
+            if not ovs_p4ctl.ovs_p4ctl_add_entry(table['switch'],table['name'], match_action):
+                self.result.addFailure(self, sys.exc_info())
+                self.fail(f"Failed to add table entry {match_action}")
+        
+        # Sleep 5 Sec before tcp connection establishment
```

Comment:
sleep for tcp establishment not required here.

Reply:
updated